Entity-ngsiv2-datamodels-schema.json

```diff
@@ -0,0 +1,238 @@
+{
+  "$schema": "http://json-schema.org/schema#",
+  "id": "https://fiware.github.io/dataModels/Entity-ngsiv2-datamodels-schema.json",
+  "title": "NGSIv2 Entity. Normalized Format for FIWARE Data Models",
+  "description": "NGSIv2 Entity. Normalized Format aligned with the usage given by FIWARE Data Models. Restrictions: geo:point, geo:line, geo:polygon not covered. keyValues representation not covered",
+  "definitions": {
+    "Identifier": {
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 256,
+      "pattern": "^[\\w\\-\\.\\{\\}\\$\\+\\*\\[\\]`|~^@!,:\\\\]+$"
+    },
+    "Name": {
```

Comment:
As far I understand this "Name" is used for entity types, attribute names, attribute types and metadata names, metadata types.

I'd suggest to have a type for each kind, given that the restrictions are different. For instance, the current schema as far as I know doesn't preclude to use "id" as attribute name (which is something that the NGSIv2 spec forbids). In addition, it forbids to use "geo:distance" as metadata name (something that would be weird but that, being strict, the NGSIv2 spec allows).

I mean, I propose to have "AttributeName", "AttributeType", "MetadataName", "MetadataType". With regards to entity type (used in "EntityFragment") I think that the "Identifier" type will be fine.

This proposal would make the schema a bit longer, but not too much and we'll gain in clearness and accuracy.

(I may be wrong. I'm not familiar with all the subtle details of JSON Schema syntax :). In that case, just tell me).

Reply:
> As far I understand this "Name" is used for entity types, attribute names, attribute types and metadata names, metadata types.
> 
> I'd suggest to have a type for each kind, given that the restrictions are different. For instance, the current schema as far as I know doesn't preclude to use "id" as attribute name (which is something that the NGSIv2 spec forbids). 


In addition, it forbids to use "geo:distance" as metadata name (something that would be weird but that, being strict, the NGSIv2 spec allows).
> 
> I mean, I propose to have "AttributeName", "AttributeType", "MetadataName", "MetadataType". With regards to entity type (used in "EntityFragment") I think that the "Identifier" type will be fine.
> 
> This proposal would make the schema a bit longer, but not too much and we'll gain in clearness and accuracy.
> 
> (I may be wrong. I'm not familiar with all the subtle details of JSON Schema syntax :). In that case, just tell me).

JMCF: You won't be able to create an attribute which name is  `id` as obviously your JSON will be invalid. Providing a rule for that would be overkilling. 

I'm in favour of keeping the simplicity of the current schema and avoid being too detailed, specially for maintenance reasons and to allow other people to easily understand the format